internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_controller.go

```diff
@@ -77,74 +74,19 @@ func NewController() (*Controller, error) {
 // Run executes all phases known to the controller in serial. `Controller`
 // aborts immediately if any phase fails.
 //nolint:gocyclo
-func (c *Controller) Run(ctx context.Context, seq runtime.Sequence, data interface{}, setters ...runtime.ControllerOption) error {
+func (c *Controller) Run(ctx context.Context, seq runtime.Sequence, data interface{}, setters ...runtime.LockOption) error {
 	// We must ensure that the runtime is configured since all sequences depend
 	// on the runtime.
-	if c.r == nil {
+	if c.R == nil {
 		return runtime.ErrUndefinedRuntime
 	}
 
-	opts := runtime.DefaultControllerOptions()
-
-	for _, f := range setters {
-		if err := f(&opts); err != nil {
-			return err
-		}
+	ctx, err := c.PriorityLock.Lock(ctx, time.Minute, seq, setters...)
+	if err != nil {
+		return err
 	}
 
-	// Allow only one sequence to run at a time with the exception of bootstrap
-	// and reset sequences.
-	switch seq { //nolint:exhaustive
-	case runtime.SequenceReset:
-		// Do not attempt to lock.
-	default:
-		if opts.Force {
-			break
-		}
-
-		if opts.Takeover {
-			c.ctxMutex.Lock()
-			if c.cancelCtx != nil {
-				c.cancelCtx()
-			}
-
-			c.ctxMutex.Unlock()
-
-			err := retry.Constant(time.Minute*1, retry.WithUnits(time.Millisecond*100)).RetryWithContext(ctx, func(ctx context.Context) error {
-				if c.TryLock() {
-					return retry.ExpectedError(fmt.Errorf("failed to acquire lock"))
-				}
-
-				return nil
-			})
-			if err != nil {
-				return err
-			}
-		} else if c.TryLock() {
-			c.Runtime().Events().Publish(&machine.SequenceEvent{
-				Sequence: seq.String(),
-				Action:   machine.SequenceEvent_NOOP,
-				Error: &common.Error{
-					Code:    common.Code_LOCKED,
```

Comment:
looks like we dropped this completely?

Reply:
it's there: https://github.com/siderolabs/talos/pull/5943/files/0ead5453fffc6414d5134b78edb6c47b1a449eeb#diff-1e941df6ba5f23ed7e30cd639a93eff4eb463ad248013757b7f13e89d0faaaf8R100